### ISSUE TITLE
Linux peer review

### ISSUE THREAD

**Issue by tt-le** [OpenLiberty/guide-microprofile-graphql#32]
Functionality
---
- [x] Check that the guide has working instructions and sample code
- [x] If any URL visits, try them on different browsers: Firefox, Chrome, Safari
- [x] If any URL visits, try `curl` command where applicable 

Formatting & Presentation
---

`README.adoc` checks:
- [x] Ensure that command substitutions use the `$(cmd)` format and not ``` `cmd` ```
- [x] Ensure that the console output of the integrated tests do not include the `[INFO]` tab
- [x] Check the quality and presentation of guide according to [Structure and Style Guideline](https://github.com/OpenLiberty/draft-guides-template/wiki/Guidelines-for-Structure-and-Styles)
- [x] Check that the pom.xml, server.xml, etc files are clean
    - [x] Remove any dependencies and features that are not required
    - [x] Ensure that the parent pom.xml files do not contain any dependencies, and all dependencies are specified in the children pom.xml files
- [x] Examples with the right outcomes are provided
- [ ] Example with the wrong outcomes, if any, are provided

`pom.xml` checks (if files are present):
- [ ] Check that the file follows the template found here: https://github.com/OpenLiberty/draft-guides-template/blob/master/finish/pom.xml
- [x] Check that the following dependencies are included in all of the `pom.xml` files:
```
<!-- Provided dependencies -->
<dependency>
  <groupId>jakarta.platform</groupId>
  <artifactId>jakarta.jakartaee-api</artifactId>
  <version>8.0.0</version>
  <scope>provided</scope>
</dependency>

<dependency>
  <groupId>org.eclipse.microprofile</groupId
  <artifactId>microprofile</artifactId>
  <version>4.0.1</version>
  <type>pom</type>
  <scope>provided</scope>
</dependency>
```
- [x]  Check that the versions of plugins are directly provided and not fed in by a file property.
  - Not formatted like so:
```
    <properties>
        <!-- Plugins -->
        <version.liberty-maven-plugin>3.2</version.liberty-maven-plugin>
    </properties>

    <!-- Liberty plugin -->
    <plugin>
        <groupId>io.openliberty.tools</groupId>
        <artifactId>liberty-maven-plugin</artifactId>
        <version>${version.liberty-maven-plugin}</version>
    </plugin>
```
  - Should be formatted like so:
```
    <!-- Liberty plugin -->
    <plugin>
        <groupId>io.openliberty.tools</groupId>
        <artifactId>liberty-maven-plugin</artifactId>
        <version>3.3.3</version>
    </plugin>
```
- [x] Check that there are 4 spaces per indent for proper formatting
- [x] Ensure that the guide is using the latest version of the `liberty-maven-plugin` or `liberty-gradle-plugin` where applicable

Overall checks:
- [x] Check the consistency of guide with the template and other guides
- [x] Check the quality of code according to the best coding practices
- [x] Check that all licensing statements are properly stated in all files, with the correct year (Should be present in all Java files + the `index.html`)
- [x] Check that the directories are properly structured
- [x] Check that some of these `page-tags` are used in a guide: `MicroProfile, Maven, Docker, Kubernetes, Gradle, Java EE, Security, Cloud`. Only these tags are visible on the website. Latest list [here](https://github.com/OpenLiberty/openliberty.io/blob/master/src/main/content/_includes/visible-tags.liquid). 
- [ ] Check the `attribution` statement is accurate for the guide
- [x] Verify the `testApp.sh` script, if any, is accurate and consistent with other guides

---

- [ ] Additional tests where applicable: 
    - [ ] Define test coverage and review with team (including guide contributor, if available)
    - [ ] Define detail test cases 
    - [ ] Consider corner cases targeting the specific guide
    - [ ] Consider corner cases UI tests
    - [ ] Consider testing URL on all browsers, ie, FF, Chrome, Safari
    - [ ] Consider testing the `curl` command for URL visits
    - [x] Make sure the index.html information is correct by visiting the root, e.g. `http://localhost:9080`
- [ ] Consider building with both Maven and Gradle build tools 
- [x] Testing with different IDEs, ie, Atom, Eclipse (Optional: VS.code, IntelliJ, Microclimate)
- [x] Run Acrolinx Checker on draft (above 70 score approximately)
- [ ] Consider SEO title and description for the guides 
- [ ] Ensure automated test is enable, set up with Travis CI, and able to schedule tests 
- [x] Run `diff -r start/ finish/` and there's no differences
- [x] Ensure that the automation tests are able to run when PR is created
- [ ] Check the appearance of the guide on test site for the following items: 
    - [x] Table of contents 
    - [x] Headings
    - [x] Paragraphs
    - [x] code snippets
    - [ ] outputs
    - [x] links
    - [x] hotspots 
- [x] Test the guide end-to-end with working instruction and sample code
- [x] Perform all the defined test cases

**Comment by tt-le** [comment]
Dependencies: `mvn versions:display-dependency-updates`
There are some new versions of dependencies available: 
I don't think we update jakartaee and junit. I also recall an issue with updating yasson.
- [ ] cxf-rt-rs-client
- [ ] cxf-rt-rs-extension-providers
- [ ] microprofile
- [ ] microprofile-graphql-api
- [ ] maybe yasson ?

```bash
[INFO] --------------------< io.openliberty.guides:system >--------------------
[INFO] Building system 1.0-SNAPSHOT                                       [2/4]
[INFO] --------------------------------[ war ]---------------------------------
[INFO] 
[INFO] --- versions-maven-plugin:2.8.1:display-dependency-updates (default-cli) @ system ---
[INFO] artifact org.apache.cxf:cxf-rt-rs-extension-providers: checking for updates from central
[INFO] artifact org.apache.cxf:cxf-rt-rs-client: checking for updates from central
[INFO] artifact org.eclipse:yasson: checking for updates from central
[INFO] artifact org.apache.httpcomponents:httpclient: checking for updates from central
[INFO] artifact org.glassfish:javax.json: checking for updates from central
[INFO] artifact org.junit.jupiter:junit-jupiter: checking for updates from central
[INFO] The following dependencies in Dependencies have newer versions:
[INFO]   jakarta.platform:jakarta.jakartaee-api ................ 8.0.0 -> 9.1.0
[INFO]   org.apache.cxf:cxf-rt-rs-client ....................... 3.4.3 -> 3.4.4
[INFO]   org.apache.cxf:cxf-rt-rs-extension-providers .......... 3.4.3 -> 3.4.4
[INFO]   org.eclipse:yasson .................................... 1.0.9 -> 2.0.2
[INFO]   org.eclipse.microprofile:microprofile ................... 4.0.1 -> 4.1
[INFO]   org.junit.jupiter:junit-jupiter .................... 5.7.1 -> 5.8.0-M1
[INFO] 
[INFO] 
[INFO] -------------------< io.openliberty.guides:graphql >--------------------
[INFO] Building graphql 1.0-SNAPSHOT                                      [3/4]
[INFO] --------------------------------[ war ]---------------------------------
[INFO] 
[INFO] --- versions-maven-plugin:2.8.1:display-dependency-updates (default-cli) @ graphql ---
[INFO] The following dependencies in Dependencies have newer versions:
[INFO]   jakarta.platform:jakarta.jakartaee-api ................ 8.0.0 -> 9.1.0
[INFO]   org.apache.cxf:cxf-rt-rs-client ....................... 3.4.3 -> 3.4.4
[INFO]   org.apache.cxf:cxf-rt-rs-extension-providers .......... 3.4.3 -> 3.4.4
[INFO]   org.eclipse:yasson .................................... 1.0.9 -> 2.0.2
[INFO]   org.eclipse.microprofile:microprofile ................... 4.0.1 -> 4.1
[INFO]   org.eclipse.microprofile.graphql:microprofile-graphql-api ...
[INFO]                                                           1.0.3 -> 1.1.0
[INFO]   org.junit.jupiter:junit-jupiter .................... 5.7.1 -> 5.8.0-M1
[INFO] 

```

**Comment by tt-le** [comment]
Acrolynx score 77: https://ibmid.acrolinx.cloud/api/v1/checking/scorecards/9011455b-5a86-43b1-b0eb-4463775beb10

Implementing system service
---
`It will include how much of the JVM heap is being used and how load the processor is under.` ->  `It will include how much of the JVM heap is being used and how much load the processor is under.` Alternatively, we could maybe rephrase it too; Something like: `It will include the JVM heap load and processor load. `

Implementing GraphQL resolvers
--- 
`It’s setup ` -> `It’s set up `

**Comment by AustinSeto** [comment]
The guide is using version `1.0.9` of the Yasson dependency, the latest `1.x` version. This was last updated at the same time as the `2.0.9` dependency. Since `2.0.9` is a major version update, I have left it on `1.0.9` for compatibility reasons.

All other dependencies mentioned are updated to their latest version. I included a minor update to the JUnit version, from `5.7.1` to `5.7.2` as well.

**Comment by tt-le** [comment]
Closing, all issues addressed.
